src/main/scala/net/ripe/rpki/publicationserver/store/fs/RrdpRepositoryWriter.scala

```diff
@@ -18,7 +18,11 @@ class RrdpRepositoryWriter extends Logging {
   }
 
   def cleanRepositoryExceptOneSession(rootDir: String, sessionId: UUID): Path = {
-    Files.walkFileTree(Paths.get(rootDir), new RemoveAllVisitorExceptOneSession(sessionId.toString))
+    Files.walkFileTree(Paths.get(rootDir), new RemoveAllVisitorExceptOneSession(sessionId.toString, None))
```

Comment:
Would it be safer (so that we do not remove a file that wasn't there at the time we checked, but created just before we delete) and simpler to always pass a timestamp here (and use `Some(System.currentTimeMillis()`) instead of `None`)? Then we could have just a timestamp, without an `Option`.

Reply:
It is indeed nicer to have only one constructor, but how is deleting FSUtil.isModifiedBefore(path, now()) safer than deleting without a condition?

Reply:
Ah, I get it now :)
Yeah, will do.